### PR TITLE
DOCS-8052, DOCS-8067 Getting Started with Monitors + Monitors Documentation Update

### PR DESCRIPTION
**Pull request by alai97**
### What does this PR do? What is the motivation?

Update Getting Started with Monitors, Monitors doc pages, rename sections with `Notify your team`, and clean up anchor links.

### Merge instructions

- [x] Please merge after reviewing

### CODE REVIEW

File: content/en/getting_started/synthetics/browser_test.md [hunk 1]
```diff
@@ -112,7 +112,7 @@ It is important to finish your browser test with an **Assertion** to ensure your
 
 {{< img src="getting_started/synthetics/record-test.mp4" alt="Record test steps" video="true" >}}
 
-The example website regularly throws an error causing it to intentionally fail. If you include your email address in the **Notify your team** field, you receive an email notification when the test fails and recovers.
+The example website regularly throws an error causing it to intentionally fail. If you include your email address in the **Configure te monitor for this test** field, you receive an email notification when the test fails and recovers.
```
Comment (alai97) [inline content/en/getting_started/synthetics/browser_test.md:115]:
```suggestion
The example website regularly throws an error causing it to intentionally fail. If you include your email address in the **Configure the monitor for this test** field, you receive an email notification when the test fails and recovers.
```

File: content/en/monitors/types/custom_check.md [hunk 2]
```diff
@@ -87,7 +87,7 @@ See the [Monitor configuration][7] documentation for information on [No data][8]
 
 ### Notifications
 
-For detailed instructions on the **Say what's happening** and **Notify your team** sections, see the [Notifications][11] page.
+For detailed instructions on the **Configure notifications and automations** sections, see the [Notifications][11] page.
```
Comment (alai97) [inline content/en/monitors/types/custom_check.md:90]:
```suggestion
For detailed instructions on the **Configure notifications and automations** section, see the [Notifications][11] page.
```

**Comment by aliciascott** [comment]
Editorial card created: https://datadoghq.atlassian.net/browse/DOCS-8080

**Review by jhgilbert — APPROVED** [review]
Approved with some minor queries/suggestions. Thanks!

**Pull request merged**